### PR TITLE
added GPTNeoXForTokenClassification

### PR DESCRIPTION
**Pull request by peter-sk**
# What does this PR do?

It adds the class GPTNeoXForTokenClassification, which allows using GPT NeoX models for token classification tasks. The implementation follows the one for other models (such as GPT2 and GPT Neo) closely and simply adds a linear layer after the hidden states.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [X] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@ArthurZucker
@younesbelkada

### CODE REVIEW

**Comment by peter-sk** [comment]
Ready for review, @ArthurZucker  and @younesbelkada  👍

**Review by younesbelkada — APPROVED** [review]
This looks good to me! Thanks for your contribution

File: src/transformers/models/gpt_neox/modeling_gpt_neox.py [hunk 1]
```diff
@@ -873,3 +878,86 @@ def forward(
...
+        super().__init__(config)
+        self.num_labels = config.num_labels
+
+        self.gpt_neox = GPTNeoXModel(config)
+        if hasattr(config, "classifier_dropout") and config.classifier_dropout is not None:
+            classifier_dropout = config.classifier_dropout
+        elif hasattr(config, "hidden_dropout") and config.hidden_dropout is not None:
+            classifier_dropout = config.hidden_dropout
+        else:
+            classifier_dropout = 0.1
```
[the remark this answers is not in the document]
Comment (peter-sk) [reply src/transformers/models/gpt_neox/modeling_gpt_neox.py:894]: The same as for GPTNeoForTokenClassification - we might just remove this and take config.classifier_dropout in both cases.

File: src/transformers/models/gpt_neox/modeling_gpt_neox.py [hunk 2]
```diff
@@ -873,3 +878,86 @@ def forward(
...
+        self.dropout = nn.Dropout(classifier_dropout)
```
Comment (sgugger) [inline src/transformers/models/gpt_neox/modeling_gpt_neox.py:895]:
```suggestion
        self.dropout = nn.Dropout(config.classifier_dropout)
```
There is a config parameter that always has a default now, so let's use it.

**Review by sgugger — APPROVED** [review]
Just one nit and this shoukd be good to merge, thanks a lot for your PR!

File: src/transformers/models/gpt_neox/modeling_gpt_neox.py (hunk 2 above)
Comment (ArthurZucker) [reply src/transformers/models/gpt_neox/modeling_gpt_neox.py:895]: Haha I said the same glad to have a validation!

Comment (peter-sk) [reply src/transformers/models/gpt_neox/modeling_gpt_neox.py:895]: Yes, I added the config parameter thanks to @ArthurZucker's feedback 👍

**Review by sgugger — APPROVED** [review]

**Pull request merged**